### DISCUSSION TITLE
DIPY Project 4: Creating Synthetic Data

### DISCUSSION THREAD

**Discussion by lb-97** [dipy/dipy#2731] [#GSoC/GSoD]
Hi,
I'm Vara Lakshmi and I'm currently pursuing Master's from NYU. I am interested in working on Project #4 - Creating Synthetic Data - as part of GSoC 2023. As a Graduate Research Assistant at NYU Video Lab and NYU Langone, I got the opportunity to study and  work with Diffusion MRI volumes. Previously I worked on Sex Classification and Cognition Prediction on HCP Young Adult Diffusion MRI dataset using Deep Learning Modeling. I developed a unique training strategy using 2D CNN to model these 3D volumes that resulted in 50% lesser training time and higher accuracy than SOTA. I would like to continue pursuing research in this field and this is a great opportunity for me to do so. I believe I can make an impactful contribution to this project, if given an opportunity.

@skoudoro @pjsjongsung

**Comment by skoudoro** [comment]
Hi @lb-97, 

Thank you for your interest in this project. You have a really interesting background. 

To get started, we recommend to take a look at the guidelines on how to contribute to DIPY.

The second step is making multiple small enhancement/bugfix/documentation fix/etc to DIPY before applying for the GSoC (mandatory). It helps you get some idea how things would work during the GSoC. The fix does not need to be related to your proposal.

Do not forget that gsoc is a program to help and bring value to opensource software.

Looking forward for your first contribution!

- If you have any specific questions concerning the gsoc project. We will be happy to answer them in this thread. 
- if you do not know what to do for contributing, you should first start to run the test, some exemples and read documentation, read all the issues. You will find  naturally what to do to help.

**Reply by lb-97** [reply]
Thankyou @skoudoro 
1. Could you please tell me if there's an existing repo for synthetic data generation. I would love to take a look and it'd help me prepare with my proposal.
2. Yes, I will start working on issues - please feel free to assign me an issue, it'd help me get started.

**Reply by pjsjongsung** [reply]
Hi @lb-97 .
Unfortunately, we do not have any existing repo yet. If it helps, our ultimate goals are conditional stable diffusion model (https://arxiv.org/abs/2112.10752) and inpainting model. Conditions will be things such as modality, damage size and location. Inpainting would be introducing tumor or lesion to a normal brain image

**Reply by lb-97** [reply]
Hi @pjsjongsung,
Thanks for the info and I apologize for the delayed response due to the hectic semester. 
I have been doing some Literature Review on Diffusion Modelling and Image Generation through DM. Here are a couple of resources I found very informative and useful - 

1. A [medium article](https://towardsdatascience.com/understanding-the-denoising-diffusion-probabilistic-model-the-socratic-way-445c1bdc5756) that discusses in plain English the design, formulation and motivation of Diffusion Modelling while explaining the [paper](https://arxiv.org/abs/2006.11239). I found it extremely useful to get into the mathematical meaning behind each component.
2. A [Blog](https://lilianweng.github.io/posts/2021-07-11-diffusion-models/) by Lilian Weng - outlining the differences between GANs, VAE and DM. Also has relevant derivations for each forward process, reverse process and Loss Function.
3. A [Blog](https://ai.googleblog.com/2021/07/high-fidelity-image-generation-using.html) by Google research - Outlining trending Diffusion Models - SR3 & Cascaded DM.


I read the paper you posted on Latent Diffusion Modelling that focuses on compute reduction and parameter reduction. It's an interesting paper with diverse applications such as inpainting and good performance on conditional image generation on ImageNet.
Here, I found other papers on [Diffusion Transformers](https://arxiv.org/pdf/2212.09748v2.pdf) and  [StyleGANs](https://arxiv.org/pdf/2202.00273v2.pdf) showing better (Conditional)Image Generation performance on ImageNet.

For our purpose i.e., Medical Image Generation/Segmentation using Diffusion Models-
1. [Medical Diffusion](https://arxiv.org/pdf/2211.03364.pdf) - It's heavily built on Denoising Diffusion Model where they replaced 2D convolutions with 3D convolutions and considered Depth as batch dimension for Attention modules.
2. [MedSigDiff](https://arxiv.org/abs/2211.00611) - Uses DM to produce medical image segmentation via conditioning on the original image at each step of reverse diffusion as well as  removing high frequency features through learnable filters to control the noise of the current step image.
3. [Brain Generation](https://link.springer.com/chapter/10.1007/978-3-031-18576-2_12) - This papers adopts LDM proposed in the [paper](https://arxiv.org/pdf/2112.10752.pdf) you mentioned. They successfully generate 100,000 3D brain images by conditioning on sex, age and brain structure.

My Literature Review shows there's active research in Medical Image Generation using Diffusion Models and GANs. Nonetheless, I couldn't find any yet that conditions on modality and damage size. Let me know your thoughts.

**Reply by lb-97** [reply]
Hi @pjsjongsung,
On this note can you kindly confirm if this internship accepts EAD OPT as a valid Visa status to work?

**Reply by pjsjongsung** [reply]
@lb-97 Thanks for the information. I was not familiar with the medical diffusion model but yes, there is no research published yet on conditioning on modality and damage, which makes the project more awesome!

I am not sure about the internship part, but let me see if I can find any information on it. I'll let you know asap. @skoudoro might know more about this.

**Reply by lb-97** [reply]
@pjsjongsung thankyou! Looking forward to it.

**Reply by pjsjongsung** [reply]
@lb-97 From what I have searched, OPT seems to be enough. I'll double check, but for now I do not think you have to worry.

**Reply by skoudoro** [reply]
Hi @lb-97 ,

This is NOT an internship. Gsoc student are recruited like an independent freelancer. 
I am not familiar with opt, but it should be ok with EAD since you can be a freelancer in EAD and having a job at the same time.

**Reply by skoudoro** [reply]
Here the link that confirm for the independent developer status: https://developers.google.com/open-source/gsoc/faq#is_gsoc_considered_an_internship_a_job_or_any_form_of_employment

**Reply by lb-97** [reply]
Thankyou guys!
